src/ctap/mod.rs

```diff
@@ -146,6 +150,52 @@ fn truncate_to_char_boundary(s: &str, mut max: usize) -> &str {
     }
 }
 
+/// Parses the metadata of an upgrade, and checks its correctness.
+///
+/// Returns the hash over the upgrade, including partition and some metadata.
+/// The metadata layout is:
+/// | -- 32B upgrade hash (SHA256) -- | 8B other |
```

Comment:
other?
Isn't there a better way to describe the purpose of these 8 bytes?

Reply:
I could go into more detail, but it's basically transparent to the application. Inside these 8B are
- 4B start address so the bootloader knows the firmware matches the location
- 4B timestamp
The exact usage of this "other" shouldn't matter from an application point of view. You want more detail in the comment?

Reply:
The full metadata content is now described here.

Reply:
Makes it easier when reading code to understand why these 8 bytes are part of the hash computation.